client/client_inventory.go

```diff
@@ -68,8 +87,26 @@ func doSubmitInventory(
 
 	r, err := api.Do(req)
 	if err != nil {
-		log.Error("Failed to submit inventory data: ", err)
-		return r, errors.Wrapf(err, "inventory submit failed")
+		if tries != 0 {
```

Comment:
I think we might as well drop this check and let `GetExponentialBackoffTime` handle it always. That would mean that we would introduce the default number of retries for everybody, but I don't think that's a problem. Agree?

Reply:
dropped, and now `GetExponentialBackoffTime` controls all.